### PR TITLE
switch to vcpkg for MSVC build

### PR DESCRIPTION
**Pull request by rfomin**
I think we can declare v10.5.1 the last version that supports Windows XP and just move on.

### CODE REVIEW

**Comment by rfomin** [comment]
Please note that the first run of vcpkg CI is very slow (about 30 minutes) because it builds everything from source. After that, it uses caching and usually takes about 2 minutes.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -253,29 +253,18 @@ if(MSVC)
...
-    if (FluidSynth_DLL_DIR)
-      file(GLOB files "${FluidSynth_DLL_DIR}/*.dll")
-      foreach(file ${files})
-        list(APPEND WOOF_DLLS ${file})
-      endforeach()
-    endif()
+    # SDL2_Mixer
+    list(APPEND WOOF_DLLS "${VCPKG_DIR}/modplug.dll")
+    list(APPEND WOOF_DLLS "${VCPKG_DIR}/ogg.dll")
+    list(APPEND WOOF_DLLS "${VCPKG_DIR}/opus.dll")
```
Comment (rfomin) [inline src/CMakeLists.txt:267]: We've used the same approach before for MSVC builds. I don't think it's worth complicating this with dynamic dependency lookup, and it's never going to be perfect. For example, what if the next version of SDL_Mixer switches to xmp with modplug?

File: CMakeLists.txt [hunk 2]
```diff
@@ -6,6 +6,11 @@ include(ExternalProject)
 # Adds the cmake directory to the CMake include path.
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
+if(WIN32)
+    set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
```
Comment (fabiangreffrath) [inline CMakeLists.txt:10]: These two lines need comments, please. I have no clue what they mean.

File: src/CMakeLists.txt (hunk 1 above)
Comment (fabiangreffrath) [reply src/CMakeLists.txt:267]: Yes, I agree. We'll always have to keep track of some of these. This isn't going to change anytime soon.

File: CMakeLists.txt (hunk 2 above)
Comment (rfomin) [reply CMakeLists.txt:13]: Fixed in https://github.com/fabiangreffrath/woof/pull/855/commits/64a5c981f43f9d0750eb6046a56ecae3c03277ff

**Comment by fabiangreffrath** [comment]
Will cross building still work after these changes?

**Comment by rfomin** [comment]
> Will cross building still work after these changes?

Should work, vcpkg is cross platform. I'll try this on a VM and report back.

**Comment by fabiangreffrath** [comment]
> Should work, vcpkg is cross platform. I'll try this on a VM and report back.

Frankly, I don't care a lot of it still works. Haven't used it anymore since the early days. But then we should remove the recipes and documentation which suggest that it is still supported.

**Comment by rfomin** [comment]
I got it to work on Ubuntu WSL, but it wasn't easy. And everywhere there are warnings that it is not tested and may break: [docs](https://github.com/microsoft/vcpkg/blob/master/docs/users/mingw.md). So I will remove it.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -78,31 +78,39 @@ jobs:
       - name: Build
         run: cmake --build build
 
-      - name: Package
```
Comment (fabiangreffrath) [inline .github/workflows/main.yml:81]: So, we'll still build MSYS2 64-bit binaries as part of the CI process, but we don't upload artifacts anymore and they won't be part of the official release binaries (i.e. these will be vcpkg builds)?

Comment (rfomin) [reply .github/workflows/main.yml:81]: > So, we'll still build MSYS2 64-bit binaries as part of the CI process, but we don't upload artifacts anymore and they won't be part of the official release binaries (i.e. these will be vcpkg builds)?

Yes, that's the plan. I'm not sure about the `lukka/run-vcpkg@v10` action, maybe I should write the vcpkg caching script myself. So this PR is still at the experimental stage.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -99,7 +99,7 @@ jobs:
           cd ${{ env.VCPKG_ROOT }}
           ./bootstrap-vcpkg.sh -disableMetrics
           nuget=$(./vcpkg.exe fetch nuget | tail -n 1)
-          owner="${GITHUB_REPOSITORY%/*}"
+          owner="rfomin"
```
Comment (rfomin) [inline .github/workflows/main.yml:104]: @fabiangreffrath Oddly enough, NuGet packages don't work on your account. I have successfully built it in my repository: https://github.com/rfomin?page=1&tab=packages Please take a look at your settings, maybe disable uploading packages? Not sure what's wrong.

Comment (fabiangreffrath) [reply .github/workflows/main.yml:104]: Me neither, I have no idea what's wrong. 🤷

Comment (fabiangreffrath) [reply .github/workflows/main.yml:104]: There is some blubber on this page 
https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -108,8 +108,8 @@ jobs:
             -storepasswordincleartext \
             -name "GitHub" \
             -username "$owner" \
-            -password "${{ secrets.GITHUB_TOKEN }}"
-          "$nuget" setapikey "${{ secrets.GITHUB_TOKEN }}" \
+            -password "${{ secrets.NUGET_ACCESS }}"
+          "$nuget" setapikey "${{ secrets.NUGET_ACCESS }}" \
```
Comment (rfomin) [inline .github/workflows/main.yml:112]: @fabiangreffrath Something went wrong, no new secret found. Please double check repository settings.

Comment (fabiangreffrath) [reply .github/workflows/main.yml:112]: That's how it looks:

Screenshot_20221229_163151_Chrome

Comment (fabiangreffrath) [reply .github/workflows/main.yml:112]: Secret updated, please try again.

Comment (rfomin) [reply .github/workflows/main.yml:110]: I sent you a new PM.

Comment (fabiangreffrath) [reply .github/workflows/main.yml]: Screenshot_20221229_181843_Chrome

Comment (rfomin) [reply .github/workflows/main.yml]: I think we should remove all these new secrets. Sorry for the misunderstanding.

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**